src/main/java/hudson/remoting/Channel.java

```diff
@@ -1389,7 +1400,25 @@ public void dumpExportTable(PrintWriter w) throws IOException {
      * @see #lastHeard
      */
     public long getLastHeard() {
-        return lastHeard;
+        // TODO - this is not safe against clock skew and is called from jenkins core (and potentially plugins)
+        try {
+            lastHeardLock.readLock().lock();
+            return lastHeard;
+        }
+        finally {
+            lastHeardLock.readLock().unlock();
+        }
+    }
+
+    private void updateLastHeard() {
+        try {
+            lastHeardLock.writeLock().lock();
+            // TODO - this is not safe against clock skew and is called from jenkins core (and potentially plugins)
+            lastHeard = System.currentTimeMillis();
```

Comment:
I don't get your point here - no one is (or should be at this point in time) waiting for the lastHeard to be updated and the RW lock is private.
All this is protecting against is one thread seeing a partial 32bit write to a 64bit number when calling getLastHeard() in a different Thread.